docs/howtos/building-a-rust-component.md

```diff
@@ -1,20 +1,179 @@
+
+### What’s the global sync ID and the collection sync ID?
+Both guids, both used to identify when the data in the server has changed
+radically underneath us (eg, when looking at lastModified is no longer a sane
+thing to do.)
+
+The "global sync ID" changing means that every collection needs to be assumed as
+having changed radically, whereas just the "collection sync ID" changing means
+just that one collection.
+
+### What’s `get_sync_assoc`, why is it important? What is `StoreSyncAssociation`?
+They are all used to track the guids above. It’s vitally important we know when
+these guids change.
+
+StoreSyncAssociation is a simple enum which reflects the state a sync engine
+can be in - either `Disconnected` (ie, we have no idea what the GUIDs are) or
+`Connected` where we know what we think the IDs are (but the server may or may
+not match with this)
+
+These GUIDs will typically be stored in the DB in the metadata table.
+
+### what is `apply_incoming` versus `sync_finished`
+`apply_incoming` is where any records incoming from the server (ie, possibly
+all records on the server if this is a first-sync, records with a timestamp
+later than our last sync otherwise) are processed.
+
+`sync_finished` is where we've done all the sync work other than uploading new
+changes to the server.
+
+### What's the diff between reset and wipe?
+
+* Reset means “I don’t know what’s on the server - I need to reconcile everything there with everything I have”. IOW, a “first sync”
+* Wipe means literally “wipe all server data”
+
+### The "engine" concept is weird, I assumed I have to implement it but it's not really relevant anymore coz since manager?
+
+Yeah, we should clean this up - but in general, you just have to implement a
+trait or 2 - all the sync machinery is already handled.
+
+# Exposing to consumers
+
+You will need an FFI or some other way of exposing stuff to your consumers.
 
-Please share your :+1: on
-[the relevant github issue](https://github.com/mozilla/application-services/issues/598)
-to let us know that you wanted it.
+We are moving towards uniffi - but this is so new that it's not even clear yet
+exactly where the canonical home for this tool is yet, let alone best practices.
 
-In the meantime you might find these earlier blog posts to be helpful:
+However, here are some earlier blog posts which might be helpful:
 
 * [Building and Deploying a Rust library on Android](https://mozilla.github.io/firefox-browser-architecture/experiments/2017-09-21-rust-on-android.html)
 * [Building and Deploying a Rust library on iOS](https://mozilla.github.io/firefox-browser-architecture/experiments/2017-09-06-rust-on-ios.html)
 * [Blog post re: lessons in binding to Rust code from iOS](https://discourse.mozilla.org/t/dear-diary-turns-out-x-platform-is-hard/25348)
 
+The above are likely to be superceded by uniffy docs, but for now, good luck!
```

Comment:
If you deliberately typo-ed this as `uniffy` in order to troll me then excellently done, I salute you 😆